### ISSUE TITLE
ESLint YAML .eslintrc file does not work

### ISSUE THREAD

**Issue by StephenGregory** [MiguelCastillo/Brackets-InteractiveLinter#153]
### Description

[ESLint supports .eslintrc being written in YAML](http://eslint.org/docs/user-guide/configuring). The Interactive Linter does not handle such a file.

### Steps to reproduce

Configure `brackets.json` appropriately:
```
{
"interactive-linter.javascript": [
        "eslint"
    ]
}
```

Create a `.eslintrc` file in your project directory with the following content:
```yaml
---
  env:
    browser: true

  rules:
    camelcase:
      - 2
```

An alert prompt displays the following:
> Error processing linter settings
> SyntaxError: Unexpected number

##### Other details
- OS: Windows 8.1
- Brackets version: 1.4
- No errors  logged in Debug > "Show Developer Tools" - "Console view".
- Other extensions:
  - Beautify
  - Brackets Git
  - Code Folding
  - Advanced New File
  - Brackets File Icons
  - CDN Finder
  - Code Outline
  - Extract for Brackets (Preview)
  - Indentator
  - Inline Regex Editor
  - QuickDocsJS
  - QuickDocsRegEx
  - Show Whitespace
  - Son of Hammsidian
  - Themes for Brackets
  - Todo
  - Tomorrow Night
  - White Space Sanitizer
  - YAML Linter

**Comment by StephenGregory** [comment]
I suspect the issue is on [this line](https://github.com/MiguelCastillo/Brackets-InteractiveLinter/blob/8e8fbb15e1d85f0637ae6c2928e5dd33582ee0a1/linterSettings.js#L73), wherein it is assumed that the settings file in in JSON.

How after deferring the parsing of the file to ESLint (see [getConfigForFile()](http://eslint.org/docs/developer-guide/nodejs-api#getconfigforfile))?

**Comment by MiguelCastillo** [comment]
@StephenGregory thanks for logging this issue.  Currently, interactive linter only supports json formatted settings files.  I am not totally against supporting YAML.  Would you be willing to put together a PR?

**Comment by StephenGregory** [comment]
Sure! For reference, how have you gone about getting the single `eslint.js` in your repo?

**Comment by PhiLhoSoft** [comment]
Yes, I have hit the issue when I switched from JSHint to ESLint, generating (with `eslint --init`) a Yaml file.
I switched to Json, but I would prefer to use Yaml as it allows comments...

BTW (slightly unrelated, I can create a new issue if needed), how can we cleanly update the ESLint version? Some options seems to be unrecognized or limited (eg. line ending and brace placement).
